uui-components/src/table/DataTableHeaderCell.tsx

```diff
@@ -25,7 +33,12 @@ export class DataTableHeaderCell<TItem, TId> extends React.Component<DataTableHe
 
     toggleSort = (e: React.MouseEvent) => {
         if (isClickableChildClicked(e) || !this.props.column.isSortable) return;
-        this.props.onSort(this.props.sortDirection === 'asc' ? 'desc' : 'asc');
+
+        let dir: SortDirection;
+        if (this.props.sortDirection === null) dir = 'asc';
```

Comment:
this condition does not catch undefined sortDirection value, may be better to use !this.props.sortDirection here?